packages/single-line/src/SingleLineEditor.tsx

```diff
@@ -72,15 +72,23 @@ export function SingleLineEditor(props: SingleLineEditorProps) {
                 setValue(e.target.value);
               }}
             />
-            {props.withCharValidation && (
+            {withCharValidation && (
               <div className={styles.validationRow}>
-                <CharCounter value={value || ''} checkConstraint={checkConstraint} />
                 <CharValidation constraints={constraints} />
+                <CharCounter
+                  value={value || ''}
+                  checkConstraint={checkConstraint}
+                  constraints={constraints}
+                />
               </div>
             )}
-            {props.withCharValidation === false && (
-              <div className={styles.validationRow}>
-                <CharCounter value={value || ''} checkConstraint={() => true} />
+            {!withCharValidation && (
+              <div className={styles.counterRow}>
+                <CharCounter
+                  value={value || ''}
+                  checkConstraint={checkConstraint}
+                  constraints={constraints}
+                />
```

Comment:
Nit: Is just the git diff or are these two sections identical with the exception of the style class and `CharValidation`? 

Maybe it would look cleaner if do something like this?

```jsx
<div className={withCharValidation? styles.validationRow: styles.counterRow}>
    <CharValidation constraints={constraints} enabled={withCharValidation}/>

    <CharCounter
      value={value || ''}
      checkConstraint={checkConstraint}
      constraints={constraints}
    />
</div>
```

Don't feel strongly about it though.

Reply:
They are, I think that is indeed cleaner